### ISSUE TITLE
xjc 2.3.3 fails to find plugins on Java 11

### ISSUE THREAD

**Issue by cwalther** [eclipse-ee4j/jaxb-ri#1474]
When run with Java 11, xjc from jaxb-ri-2.3.3 fails to find its bundled plugins: The usage text does not include the _Extensions_ section, and when called with `-Xlocator` it says `unrecognized parameter -Xlocator`. It works as expected when run with Java 8.

I can fix this by cherry-picking part of commit c1de969b3d517d68d2f8653535a8d5e65869d293 from 3.0.0-M3, specifically its change to [_jaxb-ri/bundles/xjc/src/main/java/module-info.java_](https://github.com/eclipse-ee4j/jaxb-ri/commit/c1de969b3d517d68d2f8653535a8d5e65869d293#diff-90150d3ac99ffd136fd143d97b98e8f1), adding the `provides` section. Interestingly enough, this section was once added in 39add9d139ee99106c3c6ec2f7f795fec8f7c814, but then removed again in 83c5858de344a426724749e3756785a63caa02b2. There is so much else in that latter commit that I can’t tell if that was because of a problem with it.

Is there any chance of a maintenance release in the 2.3 series with this fix? I can provide a pull request with the mentioned change if desired.

I cannot use 3.0.0 because the code it generates has renamed package names (javax.xml → jakarta.xml, com.sun.xml.bind → org.glassfish.jaxb.core). (Will we be able to turn this off in the final release? It would be useful if a new compiler could still be used to generate code for older codebases that still use javax packages. I tried `-target 2.2` (and other values) but that didn’t change anything.)

**Comment by lukasj** [comment]
short answer is that 3.0.0 needs to be ready before we can get to updating 2.3.3

as for generating javax in version 3.0.0 - it is unlikely for this to be there. We can take a look at later (it would probably make sense to create extra issue for this)

**Comment by cwalther** [comment]
That doesn’t sound too bad, thanks. I’m not in a hurry, for now my immediate itch is scratched by the locally built patched version.

I’ve split the parenthetical off into its own issue: #1475. (Of course, if that one lands, this one becomes less pressing.)

**Comment by lukasj** [comment]
fixed by 4365894 through #1512 for 2.3.4

**Comment by GFriedrich** [comment]
@lukasj: Hi, I see that the release of 2.3.4 was branched about 2 weeks ago, but I'm unable to find any artifacts on Maven Central. Are you going to publish these or does one need to create that version on his/her own?
Thanks for your help in advance.

**Comment by lukasj** [comment]
@GFriedrich It's not there yet, I still have to do some testing before considering it ready for GA. Binaries are available in https://jakarta.oss.sonatype.org/ repo but keep in my artifacts may change without prior notice there.

**Comment by GFriedrich** [comment]
@lukasj wow, thanks for the heads-up ... didn't expected that you keep me informed - thanks again!

**Comment by cwalther** [comment]
I finally got around to testing this, and it appears to work fine with jaxb-ri-2.3.4-20210219.075459-1. Thanks! Apart from changing the HTML escaping in the comments back to the way it was in 2.2 (which I assume is an improvement, but I didn’t check closely), the output in my use case is identical to my patched 2.3.3.
